airbyte-lib/tests/integration_tests/test_integration.py

```diff
@@ -58,40 +72,92 @@ def test_check_fail():
         source.check()
 
 
-def test_sync():
+def test_sync_to_duckdb(expected_test_stream_data: dict[str, list[dict[str, str | int]]]):
+    # source = ab.get_connector("source-test", config={"apiKey": "test"})
+    # cache = ab.new_local_cache(
+    #     source_catalog=source.configured_catalog,
+    # )
+
     source = ab.get_connector("source-test", config={"apiKey": "test"})
-    cache = ab.get_in_memory_cache()
+    cache = ab.new_local_cache(source_catalog=source.configured_catalog)
```

Comment:
I imagine there are two use cases, with an evolution from the first to the second over time.

At first, the user/developer will just want a local cache and not care how it is implemented. Later, the user may want to be more particular about where the data is stored and specifically how the cache is configured. The helper factory functions `new_local_cache()` and `get_default_cache()` are designed for the first use case, with "new_local_cache" biasing towards isolation of this cache from prior caches and "get_default_cache" biasing towards recycled/combined outputs because of the re-use of the same backend DB. In these cases, the user doesn't care what the backend is or how files are written, just that their data is stored somewhere and other behaviors are predictable. (DuckDB is intentionally not mentioned in the function name, and we could change implementation across versions, likely without negatively affecting the user.)

At some point, I expect the user to outgrow these functions and migrate to SnowflakeCache, PostgresCache, or similar - which have constructors designed to help with the more complex configuration required for those backends.

It feels weird to me that we might have a separate helper function for each type, like get_snowflake_cache, get_duckdb_cache, etc. Perhaps we could pass a factor function the CacheConfig object and let it decide the class based on the config object's class (or similar), but the user still needs to know a set of class names (and import them) in order to get type hinting, auto-complete, etc.

Wdyt of this approach and these relative benefits of class-based vs factory-based construction?

Reply:
Thanks for explaining, it's just something I stumbled over when looking at the code.

Your explanation makes sense to me, I think we can keep it like this, but if these classes are meant to be imported by the user IMHO we should clean up a little the exports of the `caches` module and the classes itself, as they expose a lot of stuff at the moment.